resources/Microsoft.NET.SDK.ToolInstaller/Microsoft.NET.SDK.ToolInstaller.psm1

```diff
@@ -0,0 +1,297 @@
+                throw "dotnet.exe not found in Program Files or Program Files (x86)"
+            }
+        }
+    } elseif ($IsMacOS) {
+        $dotNetPath = "/usr/local/share/dotnet/dotnet"
+        if (-not (Test-Path $dotNetPath)) {
+            $dotNetPath = "/usr/local/bin/dotnet"
+            if (-not (Test-Path $dotNetPath)) {
+                throw "dotnet not found in /usr/local/share/dotnet or /usr/local/bin"
+            }
+        }
+    } elseif ($IsLinux) {
+        $dotNetPath = "/usr/share/dotnet/dotnet"
+        if (-not (Test-Path $dotNetPath)) {
+            $dotNetPath = "/usr/bin/dotnet"
+            if (-not (Test-Path $dotNetPath)) {
+                throw "dotnet not found in /usr/share/dotnet or /usr/bin"
+            }
+        }
+    } else {
+        throw "Unsupported operating system"
+    }
+
+    Write-Verbose -Message "'dotnet' found at $dotNetPath"
+    return $dotNetPath
+}
+
+# TODO: when https://github.com/dotnet/sdk/pull/37394 is documented and version is released with option simple use --format=JSON
+
+function Convert-DotNetToolOutput {
+    [CmdletBinding()]
+    param (
+        [string[]] $Output
+    )
+
+    process {
+        # Split the output into lines
+        $lines = $Output | Select-Object -Skip 2
+
+        # Initialize an array to hold the custom objects
+        $inputObject = @()
+
+        # Skip the header lines and process each line
+        foreach ($line in $lines) {
+            # Split the line into columns
+            $columns = $line -split '\s{2,}'
+
+            # Create a custom object for each line
+            $customObject = [PSCustomObject]@{
+                PackageId = $columns[0]
+                Version   = $columns[1]
+                Commands  = $columns[2]
+            }
+
+            # Add the custom object to the array
+            $inputObject += $customObject
+        }
+
+        return $inputObject
+    }
+}
+
+function Install-DotNetToolPackage
+{
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory, ValueFromPipelineByPropertyName)]
+        [string] $Name,
+        [Parameter(ValueFromPipelineByPropertyName)]
+        [string] $Version,
+        [Parameter(ValueFromPipelineByPropertyName)]
+        [bool] $PreRelease
+    )
+    
+    begin
+    {
+        function Get-DotNetToolArguments
+        {
+            param([string]$Name, [string]$Version, [bool]$PreRelease)
+
+            $string = $Name
+            
+            if (-not ([string]::IsNullOrEmpty($Version)))
+            {
+                $string += " --version $Version"
+            }
+
+            if ($PreRelease) {
+                $string += " --prerelease"
+            }
+
+            $string += " --no-cache"
+
+            return $string
+        }
+    }
+    
+    process
+    {
+        $installArgument = Get-DotNetToolArguments @PSBoundParameters
+        $arguments = "tool install $installArgument --global --ignore-failed-sources"
```

Comment:
are all of the tools you're installing supported on the current runtimes that are installed? if not, there may be issues with running those tools.

Reply:
I agree with your statement here, however I do think users themselves should be aware what they should install. I know the architecture option is also available to be specified. It can be added later I would say. Let me know your thoughts.